packages/ui-test-sandbox/src/utils/sandbox.ts

```diff
@@ -139,44 +131,38 @@ class Sandbox {
         )
       })
     }
-
     this._observer.observe(document.body, { childList: true })
   }
 
-  async teardown() {
-    await ReactComponentWrapper.unmount()
-
+  teardown() {
+    ReactComponentWrapper.unmount()
     this._sandbox.restore()
-
     this._timeoutIds.forEach((timeoutId) => {
       clearTimeout(timeoutId)
     })
     this._timeoutIds = []
-
     this._raf.forEach((requestId) => {
       window.cancelAnimationFrame(requestId)
     })
     this._raf = []
-
     window.localStorage && window.localStorage.clear()
     window.sessionStorage && window.sessionStorage.clear()
-
     setAttributes(document.documentElement, this._attributes.document)
     setAttributes(document.body, this._attributes.body)
-
+    const newRecords = this._observer.takeRecords()
+    for (const newRecord of newRecords) {
+      this._addedNodes = this._addedNodes.concat(
+        Array.from(newRecord.addedNodes)
+      )
+    }
```

Comment:
This could fix some dangling DOM elements not unmounted. Although adding this here did not seem to have any effect :/